src/main/ruby/truffleruby/core/exception.rb

```diff
@@ -469,6 +469,28 @@ def initialize(sig, message = undefined)
     @signo = signo
     super(name_with_prefix)
   end
+
+  private
+
+  def reached_top_level
+    banned = ['VTALRM']
+    if !TruffleRuby.native?
+      banned.concat(['FPE', 'ILL', 'SEGV', 'USR1', 'QUIT'])
```

Comment:
This hardcoded list is quite unfortunate.
I think we can avoid it by looking at whether the `Signal.trap` below raises ArgumentError:
```
$ ruby -e 'Signal.trap(:USR1, "SYSTEM_DEFAULT")'
-e:1:in `trap': Signal already used by VM or OS: SIGUSR1 (ArgumentError)
```
If it does, we rescue and warn.

Could you elaborate on why should we not self-signal for VM-reserved signals?

Reply:
I still want to create a Truffle signal API, which could report to the guest language which signals are supported.

Reply:
Yeah the idea is we work it out definitely once (it's already in the docs) and then test it in that one central source of truth and everyone use that.